### ISSUE TITLE
AT 16.2, 19 targeting Client, not server; Job Creation -> Process Execution

### ISSUE THREAD

**Issue by jerstlouis** [opengeospatial/ogcapi-processes#283]
If I understand correctly, the target of the _Processes_ specification is the server (not the client).

Yet some Abstract tests seem to target the client:

A.16. point 2:

> Validate the contents of the POST request using the test [/conf/core/job-creation-request](https://docs.ogc.org/is/18-062r2/18-062r2.html#ats_core_job-creation-request).

Also in A.16 point 3 the links for `/req/core/job-creation-default-execution-mode` and `/req/core/job-creation-auto-execution-mode` are broken.

The whole requirement A.19:

> "Validate that the body of a job creation operation complies with the required structure and contents."

I also think it would be better to use the "process execution" terminlogy instead of "job creation" to reflect the fact that a job is not necessarily created when executing synchronously.

**Comment by pvretano** [comment]
@jerstlouis create a PR so that the SWG can consider the change.

B.T.W. Validating the content of a POST request is not a client requirements.  The server needs not read and validate the content of a POST request ... no?

**Comment by jerstlouis** [comment]
@pvretano These are the abstract tests. My feeling now is that these intended to be talking about the responses to these operations.

**Comment by pvretano** [comment]
@jerstlouis not at all.  Validating the input against a schema that is normatively published in the specification is a valid abstract test.

**Comment by jerstlouis** [comment]
@pvretano The way Abstract test A16.2 and A19 are written right now, it reads as if the request being **sent** to the server must be must be validated. But the test implementation (e.g. the ETS) is what is making that request.

That is why I think was is intended is to validate the **response** from the server to those operations.

So instead of "the contents of the POST request", it should say "the contents of the response to the POST operation".
And instead of "the body of a job creation operation", it should say "the body of the response to the process execution operation".

I also think that this section of the abstract test (A.2.6.1. Job Creation -- AT 16-34) is outdated compared to the requirements, because all of its links to requirements are broken. The requirements might have renamed "Job creation" to "Process execution" since those tests were written. The requirements start with `REQ_process-execute` with anchors starting with `req_core_process-execute`, whereas the ATs are named `ATS_job-creation` and still refer to `req_core_job-creation`.

**Comment by pvretano** [comment]
@jerstlouis create a PR to repair the tests as you see fit.  The SWG can then consider your PR and apply a fix if necessary.

**Comment by jerstlouis** [comment]
@pvretano Working on that PR, but in fact this also applies to [Requirement 17](https://docs.ogc.org/is/18-062r2/18-062r2.html#toc32).

> The content of a request the request body SHALL be based upon the OpenAPI 3.0 schema [execute.yaml](http://schemas.opengis.net/ogcapi/processes/part1/1.0/openapi/schemas/execute.yaml).

Currently this expresses a client requirement
(and doesn't read properly either: _a request the request body_).
We could rephrase it like the ones around saying something like:

> The server SHALL support a request body for that operation using the OpenAPI 3.0 schema [execute.yaml](http://schemas.opengis.net/ogcapi/processes/part1/1.0/openapi/schemas/execute.yaml).

But then what should the abstract test say? It seems like this is a meta-test tested by all of the following abstract tests.

**Comment by pvretano** [comment]
@jerstlouis yes, there is a a typo.  It probably should say "The content of the request body SHALL be based upon ...".  Write the PR!

**Comment by jerstlouis** [comment]
@pvretano There is a nuance between the server should validate the input vs. the server shall support a valid input.

Also the execute.yaml has no required property, and allows additional members -- so the validation would be limited to the use of those properties.

But is a server really required to completely fail a job execution if e.g. an input is not valid?
This may actually break extensions like Part 3 introducing new types of inputs.

And I am wondering what the abstract test should say -- perhaps it should refer to the following abstract test which are in fact testing the responses of such requested operation? (As opposed to an abstract test that submits wrong inputs on purpose to test that the server returns a failure).

**Comment by pvretano** [comment]
@jerstlouis update it as you see fit.  Once the PR is read we can discuss in the SWG.

**Comment by jerstlouis** [comment]
@pvretano See PR https://github.com/opengeospatial/ogcapi-processes/pull/284

**Comment by fmigneault** [comment]
> But is a server really required to completely fail a job execution if e.g. an input is not valid?

My opinion about that would be yes. I expect a server to fail a request  with an appropriate message to identify the cause rather than waste useful resources when it can be avoided, or worst, to not even give a hint about why execution might have gone wrong.
